### PR TITLE
bwbasic: init at 3.20

### PR DESCRIPTION
**Pull request by IreneKnapp**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib, stdenv, dos2unix, fetchzip }:
+
+stdenv.mkDerivation rec {
+  name = "bwbasic-3.20";
+
+  src = fetchzip {
+    url = "https://sourceforge.net/projects/bwbasic/files/bwbasic/version%203.20/bwbasic-3.20.zip/download";
+    extension = "zip";
```
Comment (veprbl) [inline pkgs/development/interpreters/bwbasic/default.nix:8]:
```suggestion
    url = "mirror://sourceforge/project/bwbasic/bwbasic/version%203.20/bwbasic-3.20.zip";
```

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  meta = with lib; {
+    description = "Bywater BASIC Interpreter";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ irenes ];
+    platforms = platforms.all;
+    homepage = "https://sourceforge.net/projects/bwbasic/";
+  };
+}
+
```
Comment (veprbl) [inline pkgs/development/interpreters/bwbasic/default.nix:37]:
```suggestion
```

**Review by veprbl — COMMENTED** [review]
I have a few minor comments

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    stripRoot = false;
+    sha256 = "08z7cl6h9cikn714ld3804bnqcpk5z4sq8vqk5gvnd04n77ayyb9";
+  };
+
+  nativeBuildInputs = [ dos2unix ];
+
+  postPatch = ''
+    dos2unix configure
+    patchShebangs configure
+    chmod 700 configure
```
Comment (legendofmiracles) [inline pkgs/development/interpreters/bwbasic/default.nix:18]:
```suggestion
    chmod +x configure
```
I find this cleaner, also makes the intent more clear

**Review by legendofmiracles — COMMENTED** [review]
Builds on my machine

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{ lib, stdenv, dos2unix, fetchzip }:
+
+stdenv.mkDerivation rec {
+  name = "bwbasic-3.20";
```
[the remark this answers is not in the document]
Comment (IreneKnapp) [reply pkgs/development/interpreters/bwbasic/default.nix]: Done.

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  postPatch = ''
+    dos2unix configure
+    patchShebangs configure
+    chmod 700 configure
+  '';
+
+  makeFlags = [
+    "CFLAGS=\"-Wno-format-security\""
+  ];
```
[the remark this answers is not in the document]
Comment (IreneKnapp) [reply pkgs/development/interpreters/bwbasic/default.nix]: Oh, cool! Didn't know about that. Done.

File: pkgs/development/interpreters/bwbasic/default.nix (hunk 1 above)
Comment (IreneKnapp) [reply pkgs/development/interpreters/bwbasic/default.nix]: Done.

File: pkgs/development/interpreters/bwbasic/default.nix (hunk 2 above)
Comment (IreneKnapp) [reply pkgs/development/interpreters/bwbasic/default.nix]: Done.

File: pkgs/development/interpreters/bwbasic/default.nix (hunk 3 above)
Comment (IreneKnapp) [reply pkgs/development/interpreters/bwbasic/default.nix]: Not my usual style, but yeah, it does seem clearer here. Done.

**Comment by IreneKnapp** [comment]
I believe I've resolved all outstanding comments, please take another look.

**Review by legendofmiracles — APPROVED** [review]

File: pkgs/development/interpreters/bwbasic/default.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, dos2unix, fetchzip }:
+
+stdenv.mkDerivation rec {
+  pname = "bwbasic";
+  version = "3.20";
+
+  src = fetchzip {
+    url = "mirror://sourceforge/project/bwbasic/bwbasic/version%203.20/bwbasic-3.20.zip";
+    extension = "zip";
```
Comment (veprbl) [inline pkgs/development/interpreters/bwbasic/default.nix:9]: The ".zip" extension is now detectable from url.

Also, you don't need to use `fetchzip`, you can use `fetchurl` and put `unzip` in `nativeBuildInputs`. The benefit of using fetchurl over fetchzip is that the https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/copy-tarballs.pl supports backing up those files to tarballs.nixos.org.

**Review by veprbl — APPROVED** [review]

File: pkgs/development/interpreters/bwbasic/default.nix (hunk 6 above)
Comment (IreneKnapp) [reply pkgs/development/interpreters/bwbasic/default.nix]: Ah thanks! Good to know. I am in favor of backups, for sure, so I've changed that.

Do you know if it's possible to make this work without writing my own unpackPhase (as the latest revision does)?

**Pull request merged**